### PR TITLE
added transfer ticket to the wallet API and updated integration tests

### LINKED ISSUE #2996: Support the `transferTicket` operation in the wallet API
After many discussions with the team, we have decided that we will be adding more operations to the Wallet API.

- Add support for the `transfer_ticket` operation to the Wallet API

Relevant links:
- https://gitlab.com/tezos/tezos/-/merge_requests/6108
- https://github.com/ecadlabs/taquito/issues/2320

### PR DESCRIPTION
**Pull request by dsawali**
closes #2996 

Thank you for your contribution to Taquito.

Before submitting this PR, please make sure:

- [x] Your code builds cleanly without any errors or warnings
- [x] You have run the linter against the changes
- [x] You have added unit tests (if relevant/appropriate)
- [x] You have added integration tests (if relevant/appropriate)
- [ ] All public methods or types have TypeDoc coverage with a complete description, and ideally an @example
- [ ] You have added or updated corresponding documentation
- [ ] If relevant, you have written a first draft summary describing the change for inclusion in Release Notes. 

In this PR, please also make sure: 

- [ ] You have linked this PR to the issue by putting `closes #TICKETNUMBER` in the description box (when applicable)
- [ ] You have added a concise description on your changes
## Release Note Draft Snippet

__If relevant, please write a summary of your change that will be suitable for
inclusion in the Release Notes for the next Taquito release.__

### CODE REVIEW

File: integration-tests/__tests__/wallet/transfer-ticket-operation.spec.ts [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+        await fundSender.confirmation();
+
+        const ticketSendOrigination = await Tezos.contract.originate({ code: ticketsSendTz, storage: null });
+        await ticketSendOrigination.confirmation();
+
+        ticketSendContract = await ticketSendOrigination.contract();
+        ticketToken = { ticketer: ticketSendContract.address, content_type: { prim: 'string' }, content: { string: 'Ticket' } };
+
+        // Send 3 tickets from the originated contract to sender
+        const sendTickets = await ticketSendContract.methods.default(senderPkh, '3').send();
```
Comment (ac10n) [inline integration-tests/__tests__/wallet/transfer-ticket-operation.spec.ts:38]: Maybe using `methodObjects` can avoid the older syntax?

Comment (dsawali) [reply integration-tests/__tests__/wallet/transfer-ticket-operation.spec.ts:38]: good point, I'll amend

**Review by hui-an-yang — APPROVED** [review]
great work!

File: integration-tests/__tests__/wallet/transfer-ticket-operation.spec.ts [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+        recipient = await createAddress();
+        sender = await createAddress();
+
+        recipientPkh = await recipient.signer.publicKeyHash();
+        senderPkh = await sender.wallet.pkh();
+
+        const fundSender = await Tezos.contract.transfer({ to: senderPkh, amount: 5 });
+        await fundSender.confirmation();
+
+        const ticketSendOrigination = await Tezos.contract.originate({ code: ticketsSendTz, storage: null });
```
Comment (hui-an-yang) [inline integration-tests/__tests__/wallet/transfer-ticket-operation.spec.ts:31]: Seeing ticketsSendTz originated here the 2nd time made me want to originate it at `integration-tests/originate-known-contracts.ts` as knownTicketContract. And `integration-tests/__tests__/rpc/nodes.spec.ts` can benefit from this too.

Comment (dsawali) [reply integration-tests/__tests__/wallet/transfer-ticket-operation.spec.ts:31]: that's not a bad idea, we can make a ticket for that.

**Pull request merged**